transport/src/main/java/io/netty/channel/DefaultChannelPipeline.java

```diff
@@ -138,13 +139,26 @@ public final Channel channel() {
         return channel;
     }
 
+    private boolean checkDestroyed(ChannelHandler handler) {
+        if (destroyed) {
+            fireExceptionCaught(
```

Comment:
not throwing an exception or any immediate indication that the insertion operation failed will be problematic for cases which assume the pipeline operation will assume reference count ownership of the handler (e.g. SniHandler inserting a SslHandler which uses the pipeline methods to manage reference count lifetime, or immediately release if exception is thrown [1]).

[1] https://github.com/netty/netty/blob/4.1/handler/src/main/java/io/netty/handler/ssl/SniHandler.java#L135